src/rewards/veterans.rs

```diff
@@ -90,6 +85,8 @@ pub fn calculate_veteran_advisors_incentives(
     total_rewards: Rewards,
     rewards_thresholds: EligibilityThresholds,
     reputation_thresholds: EligibilityThresholds,
+    agreement_rate_thresholds: impl IntoIterator<Item = Decimal> + Clone,
+    agreement_rate_modifiers: impl IntoIterator<Item = Decimal> + Clone,
```

Comment:
nit: `IntoIterator<_> + Clone` suggests me that probably we could have used a slice here and avoid extra copies